A55-stateful-session-affinity.md

```diff
@@ -0,0 +1,296 @@
+plane RPCs. We call this filter `CookieBasedStatefulSessionFilter`. It will
+copy the 3 configuration values - `name`, `ttl`  and `path` -  into the filter
+object. Validation and processing are further described in
+[Stateful session][stateful-session]. Note that
+[StatefulSessionPerRoute][stateful-session-per-route] *will* be supported.
+
+The filter is described in more detail in the section
+[`CookieBasedStatefulSessionFilter` Processing][filter-section]
+
+### Load Balancer Configuration Containing `override_host_status`
+
+Even with stateful session affinity enabled, gRPC will only send an RPC to a
+valid backend (host) as defined by
+[`common_lb_config.override_host_status`][or-host-status]. When the backend
+state is invalid, gRPC will use the configured load balancing policy to pick
+the backend.
+
+Note that gRPC currently ignores endpoints with health status other than
+`HEALTHY` or `UNKNOWN` and even `XdsClient` excludes such endpoints in its
+update to the watchers. We will need a separate design to include such
+endpoints if we determine that such behavior is needed for this feature to
+be useful to our users.
+
+As a result, for this design gRPC will only consider the `UNKNOWN` or
+`HEALTHY` states that are specified in
+[`common_lb_config.override_host_status`][or-host-status] and ignore all
+other values. We considered the alternative of NACKing (i.e. rejecting)
+the CDS update when unsupported values are present, however it was discarded
+because of the difficulty in using such configuration in mixed deployments.
+
+The `override_host_status` value is included in the config update sent to
+the `“cluster_resolver_experimental”` policy which passes down the update to
+the `“xds_cluster_impl”` policy where it is used to create the additional
+child policy as described below.
+
+### `CookieBasedStatefulSessionFilter` Processing
+
+A channel configured for stateful session affinity will have the
+`CookieBasedStatefulSessionFilter` installed (interceptor in Java/Go).
+This filter will process incoming RPCs and set an appropriate LB pick value
+which will be passed to the Load Balancing Policy's Pick method.
+
+When an RPC arrives on a configured channel the Config Selector processes it
+as described in [A31: gRPC xDS Config Selector Design][A31]. This includes the
+`CookieBasedStatefulSessionFilter` which is installed as one of the filters in
+the channel. Note that the filter maintains a context for an RPC and processes
+both the outgoing request and incoming response of the RPC within that context.
+Also note that the filter is initialized with 3 configuration values - `name`,
+`ttl`  and `path` as described above.
+
+The filter performs the following steps:
+
+* If the incoming RPC “path” does not match the `path` configuration value,
+skip any further processing and just pass the RPC through (to the next
+filter). For example if the `path` configuration value is `/Package1.Service2`
```

Comment:
However if the 'path' is "/" or "Package2.Service1/" - the filter will be processed, since it selects hostname or entire service.

Reply:
For cookie `path` matching I assume normal cookie matching rules apply (RFC6265)?

Reply:
Are you referring to https://www.rfc-editor.org/rfc/rfc6265#section-5.1.4 specifically? I'll add a reference to it (good point!)

Reply:
yes, correct.

Reply:
ref added